### PR TITLE
Fixed memory leak for NavConfig.category

### PR DESCRIPTION
**Pull request by adaybujeda**
Fixes: #2217

Possible solution for the memory leak and new tests to check for updates

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1135148780858012/1202731739707479) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by johrstrom** [comment]
Thanks for this @adaybujeda, I'm bogged down with some OSC stuff this week so there could be a delay in my seeing this.

File: apps/dashboard/app/apps/ood_app_group.rb [hunk 1]
```diff
@@ -68,6 +68,6 @@ def self.select(titles:[], groups:[])
   # then display other categories in alphabetical order
   def self.order(titles:[], groups:[])
       h = Hash[groups.map {|g| [g.title, g]}]
-      titles.concat(h.keys.sort).uniq.map { |t| h.has_key?(t) ? h[t] : nil }.compact 
+      titles.clone.concat(h.keys.sort).uniq.map { |t| h.has_key?(t) ? h[t] : nil }.compact
```
Comment (johrstrom) [inline apps/dashboard/app/apps/ood_app_group.rb:71]: Does this read better? I can't tell, but I feel like I use `concat` when I know I'm modifying the existing array, so saying `clone.concat` seems off to me.

```suggestion
      (titles + h.keys.sort).uniq.map { |t| h.has_key?(t) ? h[t] : nil }.compact
```

Comment (adaybujeda) [reply apps/dashboard/app/apps/ood_app_group.rb:71]: LGTM. Updating.

**Review by johrstrom — APPROVED** [review]
Thanks!

**Pull request merged**